codalab/model/worker_model.py

```diff
@@ -152,7 +152,10 @@ def get_workers(self):
         with self._engine.begin() as conn:
             worker_rows = conn.execute(
                 select([cl_worker, cl_worker_dependency.c.dependencies]).select_from(
-                    cl_worker.outerjoin(cl_worker_dependency)
+                    cl_worker.outerjoin(
+                        cl_worker_dependency,
+                        cl_worker.c.worker_id == cl_worker_dependency.c.worker_id,
+                    )
                 )
             ).fetchall()
             worker_run_rows = conn.execute(cl_worker_run.select()).fetchall()
```

Comment:
It is also strange to me that we do an outerjoin for the `worker_dependency` table but do this second loop for the `worker_run` table, even though they feel like they should be analogous to me. 
Git blame shows @kashizui made this change a while back. Maybe he can chime in?

Reply:
@bkgoksel I think the `worker_id` returned from `work_rows` is a superset of `worker_run_rows`. When we build the `worker_dict`, we use the superset. However, when we fill the `run_uuids` in this superset, we get that information from `worker_run_rows`. So in the end, the `get_workers()` function will return reconstructed `worker_dict` which contains all the workers from `worker` table, but only some of them will have not-none `run_uuids`.